src/main/python/covata/delta/crypto/signer.py

```diff
@@ -0,0 +1,188 @@
+#   Copyright 2017 Covata Limited or its affiliates
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from __future__ import absolute_import
+
+import json
+import re
+from base64 import b64encode
+from collections import OrderedDict
+from datetime import datetime
+
+import six.moves.urllib as urllib
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from requests.auth import AuthBase
+from collections import namedtuple
+
+from covata.delta.util import LogMixin
+
+
+class SignatureMaterial(namedtuple('SignatureMaterial', [
+    'method',
+    'uri',
+    'query_params',
+    'canonical_headers',
+    'signed_headers',
+    'hashed_payload'
+])):
+    def __init__(self, *args, **kwargs):
+        super(SignatureMaterial, self).__init__()
+        self.__canonical_request = "\n".join([
+            self.method,
+            self.uri,
+            self.query_params,
+            self.canonical_headers,
+            self.signed_headers,
+            self.hashed_payload])
+
+    @property
+    def canonical_request(self):
+        return self.__canonical_request
+
+
+class CVTSigner(AuthBase, LogMixin):
+    UNDESIRED_HEADERS = ["Connection", "Content-Length"]
+    SIGNING_ALGORITHM = "CVT1-RSA4096-SHA256"
+    CVT_DATE_FORMAT = "%Y%m%dT%H%M%SZ"
+
+    def __init__(self, crypto_service, identity_id):
+        """
+        Create a Request Signer object to sign a
```

Comment:
Should be `Creates`. Saw it in some other places too. We did the same in Java SDK. Unless it's a convention in Python to use 2nd person imperative style.